api/types/client.go

```diff
@@ -276,6 +276,12 @@ type ServiceCreateOptions struct {
 	//
 	// This field follows the format of the X-Registry-Auth header.
 	EncodedRegistryAuth string
+
+	// QueryRegistry indicates whether the service update requires
+	// contacting a registry. A registry may be contacted to retrieve
+	// the image digest and manifest, which in turn can be used to update
+	// platform or other information about the service.
+	QueryRegistry bool
```

Comment:
@tiborvass the corresponding CLI PR is https://github.com/docker/cli/pull/30.

The idea is that `QueryRegistry` should be set on all `service create` commands unless the user explicitly provides a flag to prevent it, and for all `service update` commands that update the image, using the `--image` flag (also unless the user explicitly asks to not do it with the flag).